open-grant-proposals/orbitdb-pinning/README.md

```diff
@@ -0,0 +1,194 @@
+accepts this risk and we will plan accordingly. The practical upshot of this is that we will allot
+slightly more time in our milestone schedules to account for such contingencies.
+
+### Benefits
+
+If this project succeeds, OrbitDB's users will be happily pinning their data, and
+Filecoin will have a successful example of a net-zero, non-profit pinning community
+running within its network.
+
+Also, as mentioned in the [value](#value) section above, the effort itself will raise awareness of
+Filecoin as well as help steer the roadmap based on real-world usage.
+
+## Deliverables
+
+We plan on breaking this work up into two smaller milestones, described in the next section. At
+each milestone, we will deliver the following:
+
+### Milestone 1 - Planning and Implementation
+
+- A public repository with a working implementation of OrbitDB pinning using
+the Filecoin / lotus JSON RPC APIs
+- Documentation, tests, and instructions
+
+### Milestone 2 - Launch and Outreach
+
+- One or more blog posts detailing the need and solution presented
+- Social marketing efforts like posts and chat room announcements
+- A docker image or reference implementation
+
+## Development Roadmap
+
+As per the [Risks](#risks) section, this is subject to change as the project and Filecoin
+itself mature over the coming months.
+
+Note: The budget in the milestone is based on the _equivalent_ of one full-time developer working
+on the project at a time. This metric is used for ease of understanding and the work may be the
+result of many team members via more granular chunks of work.
+
+### Milestone 1 - Implementation
+
+The implementation will be split into its back-end and front-end components.
+
+#### Milestone 1a - Back-End Filecoin Network Implementation
+
+[`orbit-db-io`] is an existing package that handles reading and writing data from the
+IPFS DAG, and is the cornerstone of [`ipfs-log`]. The grant team would repeat this
+pattern to create `orbit-db-filecoin-io` which would be similarly composable with
+[`ipfs-log`].
+
+`orbit-db-filecoin-io` would follow Filecoin standards and utilize the Filecoin / lotus
+JSON RPC APIs, allowing it to interface with any compliant Filecoin nodes, including remotely
+hosted ones.
+
+[`orbit-db-io`]: https://github.com/orbitdb/orbit-db-io
+[`ipfs-log`]: https://github.com/orbitdb/ipfs-log
+
+**Estimated duration:** 3 weeks<br />
+**Estimated cost:** 14,400€ / 16,172.14 USD
+
+#### Milestone 1b - Front-End UI Interface
+
+We will expand upon the [OrbitDB Control Center](https://github.com/orbitdb/orbit-db-control-center)
+project to build a UI to make it simple for users to take advantage of the pinning functionalities,
+and understand their own relationship in the context of the network - how much space they are
+providing, as well as how much space is available to them and their data.
+
+**Estimated duration:** 2 weeks<br />
+**Estimated cost:** 9,600€ / 10,781.42 USD
+
```

Comment:
Hey @aphelionz,

For 1b to be useful for Filecoin it needs to support setting up a FIL wallet and paying for the node transactions that the component above implements. Is that possible?

Storage deals are linked to node keys so the UI could show the node's wallet address and balance and this control center would allow users to pay FIL to that address to keep servicing orbitdb users.

Reply:
Yep! This is the general idea - whatever is required to show a working relationship between Filecoin and OrbitDB on the UI side. We'll be handling the wallet generation and transactions in 1a anyway, so this will be the UI representation.